packages/material-ui/src/ListItemText/ListItemText.js

```diff
@@ -47,8 +47,10 @@ function ListItemText(props, context) {
   } = props;
   const { dense } = context;
 
-  let primary = primaryProp || children;
-  if (primary && !disableTypography) {
+  const hasPrimary = primaryProp || primaryProp === 0;
+  let primary = hasPrimary ? primaryProp : children;
+
+  if (hasPrimary && !disableTypography) {
```

Comment:
Since `hasPrimary` is only true for `primaryProp`, this presumably won't work for `children`

Reply:
Agreed.